src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,99 @@
+
+public class Elevator extends GBSubsystem {
+
+	private final DigitalInputInputsAutoLogged digitalInputsInputs;
+	private final ElevatorCommandBuilder commandBuilder;
+	private final IRequest<Rotation2d> positionRequest;
+
+	private final ElevatorStuff elevatorStuff;
+	private final ControllableMotor frontMotor;
+	private final ControllableMotor backMotor;
+	private final IDigitalInput limitSwitch;
+
+	public Elevator(ElevatorStuff elevatorStuff) {
+		super(elevatorStuff.logPath());
+
+		this.frontMotor = elevatorStuff.frontMotorStuff().motor();
+		this.backMotor = elevatorStuff.backMotorStuff().motor();
+		this.limitSwitch = elevatorStuff.digitalInput();
+
+		this.digitalInputsInputs = new DigitalInputInputsAutoLogged();
+		this.elevatorStuff = elevatorStuff;
+		this.commandBuilder = new ElevatorCommandBuilder(this);
+		this.positionRequest = elevatorStuff.angleRequest();
+	}
+
+	public ElevatorCommandBuilder getCommandBuilder() {
+		return commandBuilder;
+	}
+
+	public void setPower(double power) {
+		frontMotor.setPower(power);
+		backMotor.setPower(power);
+	}
+
+	public void stop() {
+		frontMotor.stop();
+		backMotor.stop();
+	}
+
+	public void setBrake(boolean brake) {
+		frontMotor.setBrake(brake);
+		backMotor.setBrake(brake);
+	}
+
+	public void setTargetPosition(Rotation2d position) {
+		frontMotor.applyAngleRequest(positionRequest.withSetPoint(position));
+		backMotor.applyAngleRequest(positionRequest.withSetPoint(position));
+	}
+
+	public boolean isAtBackwardLimit() {
+		return digitalInputsInputs.debouncedValue;
+	}
+
+	public double getPositionMeters() {
+		return rotationsToMeters(
+			Rotation2d.fromRotations(
+				(elevatorStuff.frontMotorStuff().motorPositionSignal().getLatestValue().getRotations()
+					+ elevatorStuff.backMotorStuff().motorPositionSignal().getLatestValue().getRotations()) / 2
+			)
+		);
+	}
+
+	public void stayInPlace() {
+		frontMotor.applyAngleRequest(positionRequest.withSetPoint(elevatorStuff.frontMotorStuff().motorPositionSignal().getLatestValue()));
+		backMotor.applyAngleRequest(positionRequest.withSetPoint(elevatorStuff.backMotorStuff().motorPositionSignal().getLatestValue()));
+	}
+
+	public double rotationsToMeters(Rotation2d rotations) {
+		return rotations.getRotations() * ElevatorConstants.GEAR_RATIO * elevatorStuff.motorRotationsToMetersConversionRatio();
+	}
+
+	public void updateInputs() {
+		limitSwitch.updateInputs(digitalInputsInputs);
+		frontMotor.updateSignals(elevatorStuff.frontMotorStuff().motorPositionSignal(), elevatorStuff.frontMotorStuff().voltageSignal());
+		backMotor.updateSignals(elevatorStuff.backMotorStuff().motorPositionSignal(), elevatorStuff.backMotorStuff().voltageSignal());
+	}
+
+	public void logState() {
+		Logger.processInputs(elevatorStuff.digitalInputsLogPath(), digitalInputsInputs);
+		Logger.recordOutput(this.getLogPath() + "isAtBackwardLimit", isAtBackwardLimit());
+		Logger.recordOutput(this.getLogPath() + "elevatorPosition", getPositionMeters());
+	}
```

Comment:
why cant u just merge them?

Reply:
as I said in this thread, these two functions does two different things and I don't want to have sideeffects